apps/okreads-e2e/src/integration/reading-list.spec.ts

```diff
@@ -1,14 +1,76 @@
+import { $, $$, browser, By, ExpectedConditions } from 'protractor';
+
 describe('When: I use the reading list feature', () => {
-  beforeEach(() => {
-    cy.startAt('/');
+  beforeEach(async () => {
+    await browser.get('/');
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement($('tmo-root'), 'okreads')
+    );
   });
 
-  it('Then: I should see my reading list', () => {
-    cy.get('[data-testing="toggle-reading-list"]').click();
+  it('Then: I should see my reading list', async () => {
+    const readingListToggle = await $('[data-testing="toggle-reading-list"]');
+    await readingListToggle.click();
 
-    cy.get('[data-testing="reading-list-container"]').should(
-      'contain.text',
-      'My Reading List'
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement(
+        $('[data-testing="reading-list-container"]'),
+        'My Reading List'
+      )
     );
   });
-});
+
+  it('Then: I should be able to add an item to the list and then undo', async () => {
+    /* search form */
+    await $('input[type="search"]').sendKeys('Blockchain');
+    await $('form').submit();
+
+    /**
+     * adding book to reading list
+     * required to cover the edge case scenario when reading list is empty
+     */
+    await $$('[data-testing="book-add-button"][ng-reflect-disabled="false"]').first().click();
+
+    /* initial count of reading list item */
+    const readingListItem = $('tmo-total-count [ng-reflect-content]');
+    const initialItemCount = await readingListItem.getAttribute('ng-reflect-content');
+
+    /* adding another book to reading list */
+    await $$('[data-testing="book-add-button"][ng-reflect-disabled="false"]').first().click();
+
+    /* undo action */
+    await browser.driver.findElement(By.className('mat-simple-snackbar-action')).click();
```

Comment:
We don't need these browser.driver to get element, it can be done with simple query. For exmple,

```
cy.get('.mat-simple-snackbar-action button').then(($snackBarCta) => {
      // do your operation
    });
```